### PR TITLE
FetchIOCService update IocStoreConfig with feedConfigId and IOC index names

### PR DESCRIPTION
**Pull request by AWSHurneyt**
### Description
Implemented logic to update the IocStoreConfig with the saTifSourceConfig ID and IOC index names.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/org/opensearch/securityanalytics/resthandler/SATIFSourceConfigRestApiIT.java [hunk 1]
```diff
@@ -376,10 +374,7 @@ public void testRetrieveIOCsSuccessfully() throws IOException, InterruptedExcept
         }, 240, TimeUnit.SECONDS);
 
         // Confirm IOCs were ingested to system index for the feed
-        // TODO @jowg, there seems to be a bug in SATIFSourceConfigManagementService.
```
Comment (AWSHurneyt) [inline src/test/java/org/opensearch/securityanalytics/resthandler/SATIFSourceConfigRestApiIT.java:379]: @jowg-amazon will fix this in PR https://github.com/opensearch-project/security-analytics/pull/1078

File: src/main/java/org/opensearch/securityanalytics/services/STIX2IOCFeedStore.java [hunk 2]
```diff
@@ -111,10 +112,11 @@ public void storeIOCs(Map<IOC, UpdateAction> actionToIOCs) {
     }
 
     public void indexIocs(List<STIX2IOC> iocs) throws IOException {
-        // TODO @jowg, there seems to be a bug in SATIFSourceConfigManagementService.
```
Comment (AWSHurneyt) [inline src/main/java/org/opensearch/securityanalytics/services/STIX2IOCFeedStore.java:114]: @jowg-amazon will fix this in PR https://github.com/opensearch-project/security-analytics/pull/1078

File: src/test/java/org/opensearch/securityanalytics/resthandler/SATIFSourceConfigRestApiIT.java (hunk 1 above)
Comment (jowg-amazon) [reply src/test/java/org/opensearch/securityanalytics/resthandler/SATIFSourceConfigRestApiIT.java:379]: Fixed in this commit https://github.com/opensearch-project/security-analytics/pull/1078/commits/5681202abecd3cdf1cf219db4161f8744dd9da61

File: src/main/java/org/opensearch/securityanalytics/services/STIX2IOCFeedStore.java (hunk 2 above)
Comment (jowg-amazon) [reply src/main/java/org/opensearch/securityanalytics/services/STIX2IOCFeedStore.java:114]: Fixed in this commit https://github.com/opensearch-project/security-analytics/pull/1078/commits/5681202abecd3cdf1cf219db4161f8744dd9da61

File: src/main/java/org/opensearch/securityanalytics/services/STIX2IOCFeedStore.java [hunk 3]
```diff
@@ -150,7 +152,10 @@ public void indexIocs(List<STIX2IOC> iocs) throws IOException {
             long duration = Duration.between(startTime, Instant.now()).toMillis();
             STIX2IOCFetchService.STIX2IOCFetchResponse output = new STIX2IOCFetchService.STIX2IOCFetchResponse(iocs, duration);
             baseListener.onResponse(output);
-        }, baseListener::onFailure), bulkRequestList.size());
+            }, e -> {
+            log.error("Failed to index IOCs.", e);
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/services/STIX2IOCFeedStore.java:156]: can we log Tif config id

**Review by eirsep — APPROVED** [review]

File: src/main/java/org/opensearch/securityanalytics/services/STIX2IOCFeedStore.java (hunk 3 above)
Comment (AWSHurneyt) [reply src/main/java/org/opensearch/securityanalytics/services/STIX2IOCFeedStore.java]: Revised.

**Pull request merged**